### PR TITLE
gh refresh job changed

### PR DESCRIPTION
**Pull request by yzainee-zz**

### CODE REVIEW

**Comment by samuzzal-choudhury** [comment]
[test]

**Comment by miteshvp** [comment]
@yzainee - test coverage needs a drastic bump here

**Review by miteshvp — CHANGES_REQUESTED** [review]
Test coverage below threshold

**Comment by yzainee-zz** [comment]
@miteshvp earlier the test files too were under code coverage. Have removed those. Thats why you see -75% in one file. Apart from that, the github_refresh file is almost new hence we cannot compare with the older coverage %age

**Comment by miteshvp** [comment]
Does it cover all the newly added source code?

File: src/github_refresh.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+def retrieve_blob(object_key):
+    """Retrieve remote object content."""
+    return s3_resource.Object(_BUCKET, object_key).get()['Body'].read()
+
+
+def get_epv_list():
+    """Fetch the epvs from the daily stack report."""
+    if _ACCESS_KEY_ID is None or _ACCESS_KEY is None or \
```
Comment (miteshvp) [inline src/github_refresh.py]: You can use python's `any` or `all` - https://www.geeksforgeeks.org/any-all-in-python/

File: src/github_refresh.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+from selinon import run_flow_selective
+from datetime import datetime, timedelta
+from f8a_worker.setup_celery import init_celery, init_selinon
+
+import botocore
+import boto3
+import json
+
+
+_REGION = os.environ.get('AWS_S3_REGION') or 'us-east-1'
```
Comment (miteshvp) [inline src/github_refresh.py]: _REGION = os.environ.get('AWS_S3_REGION', 'us-east-1') is more pythonic

Comment (miteshvp) [reply src/github_refresh.py]: ditto for others as well

File: cico_setup.sh [hunk 3]
```diff
@@ -31,7 +31,8 @@ docker_login() {
 prep() {
     yum -y update
     yum -y install epel-release
-    yum -y install git which python36-pip python36-devel python36-virtualenv docker gcc
+    yum -y install git which python36-pip python36-devel docker gcc openssl-devel httpd-devel
+    pip3 install virtualenv pytest
```
Comment (miteshvp) [inline cico_setup.sh:35]: Is it good to install here? Or as part of runtests.sh ?

File: openshift/template.yaml [hunk 4]
```diff
@@ -43,45 +63,6 @@ objects:
...
-                      name: aws
-                      key: sqs-access-key-id
-                - name: AWS_SQS_SECRET_ACCESS_KEY
-                  valueFrom:
-                    secretKeyRef:
-                      name: aws
-                      key: sqs-secret-access-key
-                - name: PGBOUNCER_SERVICE_HOST
-                  value: bayesian-pgbouncer
-              resources:
```
Comment (miteshvp) [inline openshift/template.yaml:78]: Is this resource requirements removed by accident?

File: qa/runtests.sh [hunk 5]
```diff
@@ -7,7 +7,7 @@ pushd "${SCRIPT_DIR}/.." > /dev/null
 set -e
 set -x
 
-COVERAGE_THRESHOLD=90
+COVERAGE_THRESHOLD=80
```
Comment (miteshvp) [inline qa/runtests.sh]: it's not a great idea to change the threshold

File: qa/runtests.sh [hunk 6]
```diff
@@ -44,7 +43,7 @@ check_python_version
 
 $(which pip3) install pytest-cov
 
-PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=$(pwd) python3 "$(which pytest)" --cov=. --cov-config .coveragerc --cov-report term-missing --cov-fail-under=$COVERAGE_THRESHOLD -vv tests/
+PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=$(pwd) python3 "$(which pytest)" --cov=./src --cov-config .coveragerc --cov-report term-missing --cov-fail-under=$COVERAGE_THRESHOLD -vv tests/
```
Comment (miteshvp) [inline qa/runtests.sh:46]: there is a mechanism to exclude tests folder and other unwanted folders from test coverage

File: qa/files.txt [hunk 7]
```diff
@@ -1 +0,0 @@
-github_refresh.py
```
Comment (miteshvp) [inline qa/files.txt:1]: I guess you can add src directory here for test coverage

**Review by miteshvp — CHANGES_REQUESTED** [review]

File: qa/files.txt (hunk 7 above)
Comment (yzainee-zz) [reply qa/files.txt:1]: thats in the directory.txt. Have added it there

File: qa/runtests.sh (hunk 6 above)
Comment (yzainee-zz) [reply qa/runtests.sh:46]: was discussing this with @samuzzal-choudhury . We couldnt get any better solution. Other repos too use this

File: openshift/template.yaml (hunk 4 above)
Comment (yzainee-zz) [reply openshift/template.yaml:78]: Not doing anything with pgbouncer now in the gh refresh job. hence removed.

File: cico_setup.sh (hunk 3 above)
Comment (yzainee-zz) [reply cico_setup.sh:35]: the virtualenv is used in other scripts as well like linter, pydoc etc.

File: openshift/template.yaml (hunk 4 above)
Comment (miteshvp) [reply openshift/template.yaml:78]: You removed resources section beneath this line which needs Memory and CPU request and limit numbers

Comment (yzainee-zz) [reply openshift/template.yaml:78]: bad miss. will add it.

**Comment by yzainee-zz** [comment]
@miteshvp The covg is around 93% now. The only lines not covered one line functions which are calling either s3 function or calling selinon flow.

File: qa/runtests.sh (hunk 5 above)
Comment (yzainee-zz) [reply qa/runtests.sh]: changed it back to 90

**Review by miteshvp — APPROVED** [review]
LGTM

**Pull request merged**